http/src/request/channel/message/update_message.rs

```diff
@@ -93,38 +105,49 @@ impl<'a> UpdateMessage<'a> {
         message_id: Id<MessageMarker>,
     ) -> Self {
         Self {
+            attachment_files: None,
+            attachment_ids: None,
             channel_id,
             fields: UpdateMessageFields {
                 allowed_mentions: None,
-                attachments: &[],
+                attachments: None,
                 components: None,
                 content: None,
                 embeds: None,
                 flags: None,
+                payload_json: None,
             },
             http,
             message_id,
         }
     }
 
-    /// Specify multiple attachments already present in the target message to
-    /// keep.
+    /// Specify the [`AllowedMentions`] for the message.
     ///
-    /// If called, all unspecified attachments will be removed from the message.
-    /// If not called, all attachments will be kept.
+    /// If not called, the request will use the client's default allowed
+    /// mentions.
+    pub const fn allowed_mentions(mut self, allowed_mentions: Option<&'a AllowedMentions>) -> Self {
+        self.fields.allowed_mentions = Some(NullableField(allowed_mentions));
+
+        self
+    }
+
+    /// Attach multiple new files to the message.
     ///
-    /// Calling this method will clear any previous calls.
-    pub const fn attachments(mut self, attachments: &'a [Attachment]) -> Self {
-        self.fields.attachments = attachments;
+    /// This method clears previous calls.
+    pub const fn attachments(mut self, attachments: &'a [AttachmentFile<'a>]) -> Self {
+        self.attachment_files = Some(attachments);
 
         self
     }
 
-    /// Add multiple [`Component`]s to a message.
+    /// Set the message's list of [`Component`]s.
+    ///
+    /// Calling this method will clear previous calls.
     ///
-    /// Calling this method multiple times will clear previous calls.
+    /// # Editing
```

Comment:
do we do this style for other requests?

Reply:
yes

Reply:
<img width="284" alt="image" src="https://user-images.githubusercontent.com/1760581/149609275-7f049588-36e9-480e-b533-2d13e46f0585.png">